wallet/src/actors/worker/handlers/gen_address.rs

```diff
@@ -17,7 +17,8 @@ impl Handler<GenAddress> for worker::Worker {
         GenAddress(wallet, external, label): GenAddress,
         _ctx: &mut Self::Context,
     ) -> Self::Result {
-        self.gen_address(&wallet, external, label)
-            .map(|address| (*address).clone())
+        self.gen_address(&wallet, external, label).map(|address| {
+            (*address.expect("Address cannot be generated if wallet was never unlocked")).clone()
```

Comment:
Why `expect` in a function that returns a result? We should create a new error for this case and use `address?.clone()`.

Reply:
We cannot use `?` inside a closure... how do you propose to change it?

Reply:
Either rewrite the map as a match, or in this case you can use `and_then` instead of `map`.